File: src/Application Specific/Classes/Entity Collections/InvoiceAddress/InvoiceAddressSQL.cs
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace InvoiceAddresss
    {
        public class InvoiceAddressSQL
        {
            private Sys.Database _database;

            public InvoiceAddressSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            public void Delete(int InvoiceAddressID)
            {
                _database.ClearParameter();
                _database.AddParameter("@InvoiceAddressID", InvoiceAddressID, true);
                _database.ExecuteSP_NO_Return("InvoiceAddress_Delete");
            }

            public InvoiceAddress InvoiceAddress_Get(int InvoiceAddressID)
            {
                _database.ClearParameter();
                _database.AddParameter("@InvoiceAddressID", InvoiceAddressID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_Get");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oInvoiceAddress = new InvoiceAddress();
                        oInvoiceAddress.IgnoreExceptionsOnSetMethods = true;
                        oInvoiceAddress.SetInvoiceAddressID = dt.Rows[0]["InvoiceAddressID"];
                        oInvoiceAddress.SetAddress1 = dt.Rows[0]["Address1"];
                        oInvoiceAddress.SetAddress2 = dt.Rows[0]["Address2"];
                        oInvoiceAddress.SetAddress3 = dt.Rows[0]["Address3"];
                        oInvoiceAddress.SetAddress4 = dt.Rows[0]["Address4"];
                        oInvoiceAddress.SetAddress5 = dt.Rows[0]["Address5"];
                        oInvoiceAddress.SetPostcode = dt.Rows[0]["Postcode"];
                        oInvoiceAddress.SetTelephoneNum = dt.Rows[0]["TelephoneNum"];
                        oInvoiceAddress.SetFaxNum = dt.Rows[0]["FaxNum"];
                        oInvoiceAddress.SetEmailAddress = dt.Rows[0]["EmailAddress"];
                        oInvoiceAddress.SetSiteID = dt.Rows[0]["SiteID"];
                        oInvoiceAddress.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oInvoiceAddress.Exists = true;
                        return oInvoiceAddress;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public DataView InvoiceAddress_GetAll()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public DataView InvoiceAddress_Get_EngineerVisitID(int EngineerVisitID)
            {
                _database.ClearParameter();
                _database.AddParameter("@EngineerVisitID", EngineerVisitID, true);
                _database.AddParameter("@InvoiceEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Invoice), true);
                _database.AddParameter("@SiteEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Site), true);
                _database.AddParameter("@HQEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.HQ), true);
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_Get_EngineerVisitID");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public DataView InvoiceAddress_Get_CustomerID(int CustomerID)
            {
                _database.ClearParameter();
                _database.AddParameter("@CustomerID", CustomerID, true);
                _database.AddParameter("@InvoiceEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Invoice), true);
                _database.AddParameter("@SiteEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Site), true);
                _database.AddParameter("@HQEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.HQ), true);
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_Get_CustomerID");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public DataView InvoiceAddress_Get_SiteID(int SiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteID", SiteID, true);
                _database.AddParameter("@InvoiceEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Invoice), true);
                _database.AddParameter("@SiteEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Site), true);
                _database.AddParameter("@HQEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.HQ), true);
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_Get_SiteID");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public DataView InvoiceAddress_Get_OrderID(int OrderID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderID", OrderID, true);
                _database.AddParameter("@InvoiceEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Invoice), true);
                _database.AddParameter("@SiteEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.Site), true);
                _database.AddParameter("@HQEnumVal", Conversions.ToInteger(Sys.Enums.InvoiceAddressType.HQ), true);
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_Get_OrderID");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public InvoiceAddress Insert(InvoiceAddress oInvoiceAddress)
            {
                _database.ClearParameter();
                AddInvoiceAddressParametersToCommand(ref oInvoiceAddress);
                oInvoiceAddress.SetInvoiceAddressID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("InvoiceAddress_Insert"));
                oInvoiceAddress.Exists = true;
                return oInvoiceAddress;
            }

            public DataView InvoiceAddress_GetForSite(int SiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteID", SiteID, true);
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_GetForSite");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public DataView InvoiceAddress_Search(string criteria)
            {
                _database.ClearParameter();
                _database.AddParameter("@Criteria", criteria, true);
                var dt = _database.ExecuteSP_DataTable("InvoiceAddress_Search");
                dt.TableName = Sys.Enums.TableNames.tblInvoiceAddress.ToString();
                return new DataView(dt);
            }

            public void Update(InvoiceAddress oInvoiceAddress)
            {
                _database.ClearParameter();
                _database.AddParameter("@InvoiceAddressID", oInvoiceAddress.InvoiceAddressID, true);
                AddInvoiceAddressParametersToCommand(ref oInvoiceAddress);
                _database.ExecuteSP_NO_Return("InvoiceAddress_Update");
            }

            private void AddInvoiceAddressParametersToCommand(ref InvoiceAddress oInvoiceAddress)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@Address1", oInvoiceAddress.Address1, true);
                    withBlock.AddParameter("@Address2", oInvoiceAddress.Address2, true);
                    withBlock.AddParameter("@Address3", oInvoiceAddress.Address3, true);
                    withBlock.AddParameter("@Address4", oInvoiceAddress.Address4, true);
                    withBlock.AddParameter("@Address5", oInvoiceAddress.Address5, true);
                    withBlock.AddParameter("@Postcode", oInvoiceAddress.Postcode, true);
                    withBlock.AddParameter("@TelephoneNum", oInvoiceAddress.TelephoneNum, true);
                    withBlock.AddParameter("@FaxNum", oInvoiceAddress.FaxNum, true);
                    withBlock.AddParameter("@EmailAddress", oInvoiceAddress.EmailAddress, true);
                    withBlock.AddParameter("@SiteID", oInvoiceAddress.SiteID, true);
                }
            }


            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/InvoiceAddress/InvoiceAddressSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace InvoiceAddresss

        Public Class InvoiceAddressSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"


            Public Sub Delete(ByVal InvoiceAddressID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@InvoiceAddressID", InvoiceAddressID, True)
                _database.ExecuteSP_NO_Return("InvoiceAddress_Delete")
            End Sub

            Public Function [InvoiceAddress_Get](ByVal InvoiceAddressID As Integer) As InvoiceAddress
                _database.ClearParameter()
                _database.AddParameter("@InvoiceAddressID", InvoiceAddressID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_Get")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oInvoiceAddress As New InvoiceAddress
                        With oInvoiceAddress
                            .IgnoreExceptionsOnSetMethods = True
                            .SetInvoiceAddressID = dt.Rows(0).Item("InvoiceAddressID")
                            .SetAddress1 = dt.Rows(0).Item("Address1")
                            .SetAddress2 = dt.Rows(0).Item("Address2")
                            .SetAddress3 = dt.Rows(0).Item("Address3")
                            .SetAddress4 = dt.Rows(0).Item("Address4")
                            .SetAddress5 = dt.Rows(0).Item("Address5")
                            .SetPostcode = dt.Rows(0).Item("Postcode")
                            .SetTelephoneNum = dt.Rows(0).Item("TelephoneNum")
                            .SetFaxNum = dt.Rows(0).Item("FaxNum")
                            .SetEmailAddress = dt.Rows(0).Item("EmailAddress")
                            .SetSiteID = dt.Rows(0).Item("SiteID")

                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oInvoiceAddress.Exists = True
                        Return oInvoiceAddress
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function [InvoiceAddress_GetAll]() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function

            Public Function InvoiceAddress_Get_EngineerVisitID(ByVal EngineerVisitID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@EngineerVisitID", EngineerVisitID, True)
                _database.AddParameter("@InvoiceEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Invoice), True)
                _database.AddParameter("@SiteEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Site), True)
                _database.AddParameter("@HQEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.HQ), True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_Get_EngineerVisitID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function

            Public Function InvoiceAddress_Get_CustomerID(ByVal CustomerID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@CustomerID", CustomerID, True)
                _database.AddParameter("@InvoiceEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Invoice), True)
                _database.AddParameter("@SiteEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Site), True)
                _database.AddParameter("@HQEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.HQ), True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_Get_CustomerID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function

            Public Function InvoiceAddress_Get_SiteID(ByVal SiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@SiteID", SiteID, True)
                _database.AddParameter("@InvoiceEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Invoice), True)
                _database.AddParameter("@SiteEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Site), True)
                _database.AddParameter("@HQEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.HQ), True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_Get_SiteID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function

            Public Function InvoiceAddress_Get_OrderID(ByVal OrderID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@OrderID", OrderID, True)
                _database.AddParameter("@InvoiceEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Invoice), True)
                _database.AddParameter("@SiteEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.Site), True)
                _database.AddParameter("@HQEnumVal", CInt(Entity.Sys.Enums.InvoiceAddressType.HQ), True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_Get_OrderID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function

            Public Function [Insert](ByVal oInvoiceAddress As InvoiceAddress) As InvoiceAddress
                _database.ClearParameter()
                AddInvoiceAddressParametersToCommand(oInvoiceAddress)

                oInvoiceAddress.SetInvoiceAddressID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("InvoiceAddress_Insert"))
                oInvoiceAddress.Exists = True
                Return oInvoiceAddress
            End Function

            Public Function InvoiceAddress_GetForSite(ByVal SiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@SiteID", SiteID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_GetForSite")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function

            Public Function [InvoiceAddress_Search](ByVal criteria As String) As DataView
                _database.ClearParameter()
                _database.AddParameter("@Criteria", criteria, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("InvoiceAddress_Search")
                dt.TableName = Entity.Sys.Enums.TableNames.tblInvoiceAddress.ToString
                Return New DataView(dt)
            End Function


            Public Sub [Update](ByVal oInvoiceAddress As InvoiceAddress)
                _database.ClearParameter()
                _database.AddParameter("@InvoiceAddressID", oInvoiceAddress.InvoiceAddressID, True)
                AddInvoiceAddressParametersToCommand(oInvoiceAddress)
                _database.ExecuteSP_NO_Return("InvoiceAddress_Update")
            End Sub



            Private Sub AddInvoiceAddressParametersToCommand(ByRef oInvoiceAddress As InvoiceAddress)
                With _database
                    .AddParameter("@Address1", oInvoiceAddress.Address1, True)
                    .AddParameter("@Address2", oInvoiceAddress.Address2, True)
                    .AddParameter("@Address3", oInvoiceAddress.Address3, True)
                    .AddParameter("@Address4", oInvoiceAddress.Address4, True)
                    .AddParameter("@Address5", oInvoiceAddress.Address5, True)
                    .AddParameter("@Postcode", oInvoiceAddress.Postcode, True)
                    .AddParameter("@TelephoneNum", oInvoiceAddress.TelephoneNum, True)
                    .AddParameter("@FaxNum", oInvoiceAddress.FaxNum, True)
                    .AddParameter("@EmailAddress", oInvoiceAddress.EmailAddress, True)
                    .AddParameter("@SiteID", oInvoiceAddress.SiteID, True)
                End With
            End Sub


#End Region

        End Class

    End Namespace

End Namespace


